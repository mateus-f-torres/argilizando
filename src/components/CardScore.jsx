//@flow
import * as React from 'react';
import type {Score} from '../types/index.js';

type Props = Score & {
  handleClick: (SyntheticEvent<HTMLButtonElement>) => void,
};

const btn = 'score-btn waves-effect waves-light';

const CardScore = (props: Props) => (
  <li>
    <h4>{props.long}</h4>
    <h5 name={props.short}>
      <span className="score">{props.score}</span>
      <span className="mod">{props.mod > 0 ? '+' + props.mod : props.mod}</span>
      Next lvl cost : <span className="cost">{props.cost}</span>
    </h5>
    <button onClick={props.handleClick} className={btn} name="+">+</button>
    <button onClick={props.handleClick} className={btn} name="-">-</button>
    <button onClick={props.handleClick} className={btn} name="?">?</button>
    {props.show && <p>{props.show}</p>}
  </li>
);

export default CardScore;
