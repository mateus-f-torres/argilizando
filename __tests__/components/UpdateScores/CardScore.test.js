import React from 'react';
import {shallow} from 'enzyme';
import CardScore from 'Components/UpdateScores/CardScore.js';

describe('<CardScore />', () => {
  describe('test handleClick', () => {
    const mockScore = {
      long: 'Strength',
      score: 8,
      mod: -1,
      cost: 1,
      show: false,
    };

    test('call handleClick with "+"', () => {
      const mockHandleClick = jest.fn();

      const wrapper = shallow(
        <CardScore handleClick={mockHandleClick} {...mockScore}/>);

      wrapper.find('[data-test="score-plus"]').simulate('click');
      expect(mockHandleClick.mock.calls.length).toBe(1);
    });

    test('call handleClick with "-"', () => {
      const mockHandleClick = jest.fn();

      const wrapper = shallow(
        <CardScore handleClick={mockHandleClick} {...mockScore}/>);

      wrapper.find('[data-test="score-minus"]').simulate('click');
      expect(mockHandleClick.mock.calls.length).toBe(1);
    });

    test('call handleClick with "?"', () => {
      const mockHandleClick = jest.fn();

      const wrapper = shallow(
        <CardScore handleClick={mockHandleClick} {...mockScore}/>);

      wrapper.find('[data-test="score-show"]').simulate('click');
      expect(mockHandleClick.mock.calls.length).toBe(1);
    });
  });

  describe('"maybe-there" props tests', () => {
    const mockScoreA = {
      long: 'Dexterity',
      score: 12,
      mod: 1,
      cost: 2,
      show: false,
    };
    const mockScoreB = {
      long: 'Charisma',
      score: 6,
      mod: -2,
      cost: 1,
      show: 'mock text',
    };

    test('concat "+" with above than 0 scores', () => {
      const wrapper = shallow(<CardScore {...mockScoreA}/>);

      expect(wrapper.find('.mod').text()).toMatch(/^\+1$/);
    });

    test('dont concat "+" with 0 or lower scores', () => {
      const wrapper = shallow(<CardScore {...mockScoreB}/>);

      expect(wrapper.find('.mod').text()).toMatch(/^-2$/);
    });

    test('if (props.show) render text', () => {
      const wrapper = shallow(<CardScore {...mockScoreB}/>);

      expect(wrapper.find('[data-test="score-text"]')
        .exists()).toBe(true);
    });
    test('if (!props.show) dont render text', () => {
      const wrapper = shallow(<CardScore {...mockScoreA}/>);

      expect(wrapper.find('[data-test="score-text"]')
        .exists()).toBe(false);
    });
  });

  describe('Snapshot test', () => {
    const mockScore = {
      long: 'Wisdom',
      short: 'wis',
      score: 10,
      mod: 0,
      cost: 1,
      show: 'mock text',
    };

    const wrapper = shallow(<CardScore {...mockScore} />);

    expect(wrapper).toMatchSnapshot();
  });
});
