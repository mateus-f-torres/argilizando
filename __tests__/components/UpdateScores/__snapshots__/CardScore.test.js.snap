// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<li>
  <h4>
    Wisdom
  </h4>
  <h5
    name="wis"
  >
    <span
      className="score"
    >
      10
    </span>
    <span
      className="mod"
    >
      0
    </span>
    Next lvl cost : 
    <span
      className="cost"
    >
      1
    </span>
  </h5>
  <button
    className="score-btn waves-effect waves-light"
    data-test="score-plus"
    name="+"
  >
    +
  </button>
  <button
    className="score-btn waves-effect waves-light"
    data-test="score-minus"
    name="-"
  >
    -
  </button>
  <button
    className="score-btn waves-effect waves-light"
    data-test="score-show"
    name="?"
  >
    ?
  </button>
  <p
    data-test="score-text"
  >
    mock text
  </p>
</li>
`;
